tests/endpoints/test_assign_qc_state.py

```diff
@@ -199,8 +199,8 @@ def test_error_on_unclaimed_well(test_client: TestClient, test_data_factory):
     )
 
 
-def test_error_on_preclaimed_well(test_client: TestClient, test_data_factory):
-    """Test error on assigning a new state to a well claimed by another user."""
+def test_preclaimed_well(test_client: TestClient, test_data_factory):
+    """Other valid users should be able to set new QC states."""
```

Comment:
It's good to keep the test. The test will be more convincing is we try to assign a state that is different from the current state of the entity, for example'Failed'

Reply:
Altered to Fail as you suggest.